table/block.h

```diff
@@ -269,6 +269,9 @@ class BlockIter final : public InternalIterator {
     current_ = restarts_;
     status_ = s;
 
+    // Call cleanup callbacks.
+    Cleanable::Reset();
```

Comment:
This is the fix. In my runs of the regression tests this consistently makes the regression go away.

What was probably going on is that BlockBasedTableIterator didn't unpin blocks in block cache as it transitioned from block to block. Instead it unpinned all at once after finishing a file. This took up a lot of space in block cache, leaving less for the `Get()`s.